s4e-backend/src/main/java/pl/cyfronet/s4e/admin/sync/api/SyncJobMapper.java

```diff
@@ -0,0 +1,40 @@
+package pl.cyfronet.s4e.admin.sync.api;
+
+import org.mapstruct.Mapper;
+import pl.cyfronet.s4e.admin.sync.task.SyncJob;
+import pl.cyfronet.s4e.config.MapStructCentralConfig;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Mapper(config = MapStructCentralConfig.class)
+public abstract class SyncJobMapper {
+    public abstract SyncJobResponse toResponse(SyncJob syncJob);
+
+    public abstract SyncJobResponse.Extended toExtendedResponse(SyncJob syncJob);
+
+    private SyncJobResponse.StateDetailsResponse toResponse(SyncJob.StateDetails stateDetails) {
+        if (stateDetails instanceof SyncJob.PendingDetails) {
```

Comment:
Won't be more readable to split this method like this?

```
toResponse() {
  val stateDetails = getDetails(...);
  return stateDetails != null ? toResponse(stateDetail) : null;
}

getDetails() {
switch (typeof stateDetails) {
  case SyncJob.PendingDetails:
    return SyncJob.PendingDetails;
 ... 
  default:
    return null;
}
}

Reply:
Unfortunately, it's not possible in Java due to static linking of overloaded methods, see for example https://stackoverflow.com/questions/4523874/is-dynamic-overload-resolution-possible-in-java.
The idea is considered by the MapStruct community though, see issue 131 in their repo (I'm not linking so it doesn't show up in their issue)